routing/unified_edges.go

```diff
@@ -62,9 +63,16 @@ func (u *nodeEdgeUnifier) addPolicy(fromNode route.Vertex,
 		u.edgeUnifiers[fromNode] = unifier
 	}
 
+	// We make sure that if no payload size function is supplied we default
+	// to the payload size function of a non-blinded intermediate hop.
+	if payloadSize == nil {
```

Comment:
I'd say to not overwrite it here to make sure we catch a misuse of this method.

Reply:
good point, I think this is better

Reply:
Now triggering a graceful shutdown if this happens, but currently unsure how to test this behaviour do you maybe have ideas?

Reply:
Sadly no easy way to test this unless you pass the logger into the function (not sure it's worth it). You _could_ try to override the global `log` variable with a mock that you can assert that we called `Critical` on, but I think it's okay if this part isn't covered tbh.

Reply:
Let's see if we can find consensus in regards of not testing this code path, I left it untested for now.